src/pip/_internal/index.py

```diff
@@ -1313,6 +1313,9 @@ def _get_encoding_from_headers(headers):
     return None
 
 
+_safe_chars_re = re.compile(r'([/@]|%2F|%40)', re.I)
```

Comment:
I would add a comment above, like

```
# percent-encoded:                   /   @
_safe_chars_re = re.compile(r'([/@]|%2F|%40)', re.I)
```

Reply:
I never would've thought of doing this, and this is a good idea! ^>^